File: 633-sum-of-square-numbers/sum-of-square-numbers.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        a=0
        b=int(sqrt(c))
        while a<=b:
            temp1=a*a
            temp2=b*b
            if temp1+temp2==c:
                return True
            elif temp1+temp2>c:
                b-=1
            else:
                a+=1
        return False

File: 633-sum-of-square-numbers/sum-of-square-numbers.cpp
class Solution {
public:
    bool judgeSquareSum(int c) {
        long long low=0;
        long long high=sqrt(c);
        while(low<=high){
            long long a=low*low;
            long long b=high*high;
            if(a+b==c) return true;
            else if((a+b)>c) high--;
            else{
                low++;
            }
        }
        return false;
        
    }
};